packages/react-form/README.md

```diff
@@ -698,88 +731,65 @@ You can choose to initialize the list with an existing number of cards or no car
-          onClick={() => moveItem(index, index - 1)}
-        >
-          Move Item Up
-        </Button>
-      </div>
-      <div>
-        <Button
-          disabled={index === fields.length}
-          onClick={() => moveItem(index, index + 1)}
-        >
-          Move Item Down
-        </Button>
-      </div>
-      <Button disabled={!dirty} onClick={reset}>
-        Reset
+      <Button onClick={() => removeItem(index)}>Remove</Button>
+      <Button disabled={index === 0} onClick={() => moveItem(index, index - 1)}>
+        Move Item Up
       </Button>
-    </FormLayout.Group>
-  ));
-}
-<Button onClick={() => addItem()}>Add Card</Button>;
-```
-
-We render our UI representation of the fields by utilizing the `fields.map` function. For each field, we can use the handlers such as onChange, value, onBlur. These are the same handlers that useList provides.
-
-We can also utilize the `removeItem`, and `addItem` functions. In this example, these functions are attached to a button press. When we remove an item we pass in the index (to indicate what field item to remove). When we add a item, we utilize the factory we passed in when initializing `useDynamicList`.
-
-Note that addItem can also take in an argument which can then be passed onto the factory. In this case, the factory can be
-
-```tsx
-const emptyCardFactory = (factoryArgument: any): Card => ([{
-  cardNumber: '',
-  cvv: '',
-});
+      <Button
+        disabled={index === fields.length - 1}
+        onClick={() => moveItem(index, index + 1)}
+      >
+        Move Item Down
+      </Button>
+      <hr />
+    </div>
+  ))}
+  <Button onClick={() => addItem()}>Add Card</Button>
+  <Button disabled={!dirty} onClick={reset}>
+    Reset
+  </Button>
+</FormLayout>
 ```
 
-You can then use this argument to do as you wish :).
-
-#### Does this work with useForm?
+##### How to use it with `useForm`
 
-Yes this works out of the box with `useForm` and is compatible with what `useList` is compatible with.
+You can use `useDynamicList` with `useForm` in two ways:
 
-You would utilize it the following way in order to use the forms reset and dirty state for the `DynamicList` as well as others fields.
+You can directly pass one or several dynamic lists within `fields`. But note that the `reset` function returned by `useForm` won't reset the dynamic list to its initial value. It will only reset the value of each field for each item. So for example if you remove an item, clicking on reset won't re-add it. Plus the adding or removing a field won't make the form dirty. The form will only be dirty if you edit the value of one item's field.
 
 ```tsx
-const {submit, dirty, submitting, reset} = useForm({
+const form = useForm({
   fields: {
-    title: useField('')
-  },
-  dynamicLists: {
-    customerCards: useDynamicList<Card>([{cardNumber: '4242 4242 4242 4242', cvv: '422'}], emptyCardFactory)
+    customerCards: fields, // the fields destructured from useDynamicList
```

Comment:
I would add the `useDynamicList` detructuring above, it would be easier to understand what we mean here.